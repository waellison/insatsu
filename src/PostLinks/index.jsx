import React from "react"

const PostLinks = (props) => {
    let {next, prev} = props
    let nextLink = undefined
    let prevLink = undefined

    if(prev) {
        prevLink = <a className="prevLink" href={`/posts/${prev}`}>Previous Post</a>
    }

    if(next) {
        nextLink = <a className="nextLink" href={`/posts/${next}`}>Next Post</a>
    }

    return (
        <p>
            {prevLink} {(prev && next) ? " â€¢ " : ""} {nextLink}
        </p>
    )
}

export default PostLinks
