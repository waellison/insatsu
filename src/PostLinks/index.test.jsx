import React from 'react';
import renderer from 'react-test-renderer';
import PostLinks from './index';

test('PostLinks renders as expected with prev/next links', () => {
    const navBar = renderer.create(<PostLinks prev={7} next={9}/>);
    let tree = navBar.toJSON();
    expect(tree).toMatchSnapshot();
});

test('PostLinks renders as expected with next link only', () => {
    const navBar = renderer.create(<PostLinks next={9}/>);
    let tree = navBar.toJSON();
    expect(tree).toMatchSnapshot();
});

test('PostLinks renders as expected with prev link only', () => {
    const navBar = renderer.create(<PostLinks prev={9}/>);
    let tree = navBar.toJSON();
    expect(tree).toMatchSnapshot();
});
