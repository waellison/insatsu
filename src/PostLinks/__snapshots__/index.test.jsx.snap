// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostLinks renders as expected with next link only 1`] = `
<p>
   
   
  <a
    className="nextLink"
    href="/posts/9"
  >
    Next Post
  </a>
</p>
`;

exports[`PostLinks renders as expected with prev link only 1`] = `
<p>
  <a
    className="prevLink"
    href="/posts/9"
  >
    Previous Post
  </a>
   
   
</p>
`;

exports[`PostLinks renders as expected with prev/next links 1`] = `
<p>
  <a
    className="prevLink"
    href="/posts/7"
  >
    Previous Post
  </a>
   
   â€¢ 
   
  <a
    className="nextLink"
    href="/posts/9"
  >
    Next Post
  </a>
</p>
`;
